instrumentation/spring/spring-boot-resources/library/src/main/java/io/opentelemetry/instrumentation/spring/resources/SpringBootServiceNameDetector.java

```diff
@@ -160,6 +161,14 @@ private String findByClasspathApplicationYaml() {
     return result;
   }
 
+  @Nullable
+  private String findByBootInfApplicationYml() {
+    String result =
+        loadFromBootInf("application.yml", SpringBootServiceNameDetector::parseNameFromYaml);
```

Comment:
Looks like you've found a bug in the `findByClasspath*()` methods - they both should in fact be loading files from the `BOOT-INF/classes` directory. Can you fix these two instead of adding this new method?

Reply:
To me this looks wrong. @adamleantech why do yo need to explicitly mention `BOOT-INF/classes`, spring boot class loader automatically handles finding stuff from `BOOT-INF/classes` you shouldn't need to add it to resource path.

Reply:
the java agent executes this code before the spring boot app has had chance to do this

Reply:
@mateuszrzeszutek I found the existing functionality useful when testing the agent against spring boot apps in my IDE where the code has not been assembled into a JAR and therefore the resources are in the usual classpath resource location.  Maybe findByClasspath*() should be updated to look in both places?

Reply:
Hey @adamleantech I just merged #8101; can you merge/rebase your PR with `main`?